hw/ip/otbn/dv/smoke/smoke_test.s

```diff
@@ -125,13 +128,18 @@ csrrs x23, mod5, x0
 # w1 = mod = 0x78fccc06_2228e9d6_89c9b54f_887cf14e_c79af825_69be586e_9866bb3b_53769ada
 bn.wsrr w1, 0x0 /* MOD */
 
-# Request an RND value with a write to CSR RND_PREFETCH
-csrrw x0, rnd_prefetch, x0
-
-# sim environment provides a fixed value for RND (in other environment RND isn't
-# fixed so this test will have a different final state)
 # w2 = rnd = 0xAAAAAAAA_99999999_AAAAAAAA_99999999_AAAAAAAA_99999999_AAAAAAAA_99999999
-bn.wsrr w2, 0x1 /* RND */
+.ifdef deterministic
```

Comment:
I suggest you add a `nop` to the "else branch". That way, the instruction addresses will be unchanged by the value of `deterministic`.

Similarly, I'd suggest sticking a bunch of `nop`s on the other side for the stuff below where there's `.ifnotdef deterministic`.

Reply:
I don't think any `nop`s are needed. The address changes was due to setting a `.text` section.